python/semantic_kernel/memory/storage/chroma_data_store.py

```diff
@@ -0,0 +1,164 @@
+                    chroma_db_impl="duckdb+parquet", persist_directory=persist_directory
+                )
+        self._client = chromadb.Client(self._client_settings)
+        self._persist_directory = persist_directory
+        self._default_embdding_function = ef.SentenceTransformerEmbeddingFunction()
+        self._default_query_includes = ["embeddings", "metadatas", "documents"]
+
+    async def get_collection_async(self, collection: str) -> Optional["Collection"]:
+        try:
+            # TODO: ChromeDB reject camel case collection names.
+            #   need to decide ChromaDataStore will do auto conversion inside or user takes responseibilities
+            collection_snake_case = camel_to_snake(collection)
+            return self._client.get_collection(
+                name=collection_snake_case,
+                embedding_function=self._default_embdding_function,
+            )
+        except ValueError:
+            return None
+
+    async def get_collections_async(self) -> List[str]:
+        return [collection.name for collection in self._client.list_collections()]
+
+    async def get_all_async(self, collection: str) -> List[DataEntry]:
+        collection = self.get_collection_async(collection)
+        if collection is None:
+            return []
+
+        values = collection.get(include=self._default_query_includes)
+        records = self.query_results_to_memory_records(values)
+
+        entries = [
+            DataEntry(
+                key=values["ids"][i][0],
+                value=record,
+                timestamp=datetime.fromtimestamp(
+                    values["metadatas"][i][0]["timestamp"]
+                ),
+            )
+            for i, record in enumerate(records)
+        ]
+
+        return entries
+
+    async def get_async(self, collection: str, key: str) -> Optional[DataEntry]:
+        collection = self._client.get_collection(name=collection)
+        if collection is None:
+            return None
+
+        value = collection.get(ids=key, include=self._default_query_includes)
+        record = self.query_results_to_memory_records(value)[0]
+        timestamp = datetime.fromtimestamp(value["metadatas"][0]["timestamp"])
+
+        return DataEntry(key=key, value=record, timestamp=timestamp)
+
+    async def put_async(self, collection: str, value: DataEntry) -> DataEntry:
+        record: MemoryRecord = value.value
+        collection: Collection = self._client.get_or_create_collection(
+            name=camel_to_snake(collection),
+            embedding_function=self._default_embdding_function,
+        )
+        collection.add(
+            metadatas={
+                "timestamp": value.timestamp.timestamp(),
+                "is_reference": record.is_reference,
+                "external_source_name": record.external_source_name or "",
+                "description": record.description or "",
+            },
+            embeddings=record.embedding.tolist(),
```

Comment:
@awharrison-28 this line forces chroma use semantic-kernel's embedding

Reply:
you can find detailed behavior information [here](https://docs.trychroma.com/usage-guide)

> Alternatively, you can supply a list of document-associated embeddings directly, and Chroma will store the associated documents without embedding them itself.
```
await collection.add(
    documents=["doc1", "doc2", "doc3", ...],
    **embeddings=[[1.1, 2.3, 3.2], [4.5, 6.9, 4.4], [1.1, 2.3, 3.2], ...],**
    metadatas=[{"chapter": "3", "verse": "16"}, {"chapter": "3", "verse": "5"}, {"chapter": "29", "verse": "11"}, ...],
    ids=["id1", "id2", "id3", ...]
```

Reply:
if embeddings are not provided for `collection.add`, chroma will generate embedding with default embedding method.

Reply:
i didn't expose _default_embedding_function because I thought `ChromaDataStore` will always get embeddings from semantic-kernel. but if you think DataStore need to generate embedding itself, this should be changed :)

Reply:
Hm ok, I'll run ChromaMemoryStore with notebook 6 again. I the similarity scores I was seeing didn't match my expectations for OpenAI's text-embedding-ada-002 model, and I saw that a model was downloaded while setting up the MemoryStore.

Reply:
> > i didn't expose _default_embedding_function because I thought `ChromaDataStore` will always get embeddings from semantic-kernel. but if you think DataStore need to generate embedding itself, this should be changed :)
> 
> Nope, the DataStore should not be generating embeddings, the intended behavior should stay as-is

Yup I think we are on the same line. Datastore should not generates embeddings :). I'll give details while writing documentation.